Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp

```diff
@@ -212,17 +212,48 @@ void ReadableStreamDefaultReader::read_all_bytes(GC::Ref<ReadLoopReadRequest::Su
     readable_stream_default_reader_read(*this, read_request);
 }
 
-void ReadableStreamDefaultReader::read_all_chunks(GC::Ref<ReadLoopReadRequest::ChunkSteps> chunk_steps, GC::Ref<ReadLoopReadRequest::SuccessSteps> success_steps, GC::Ref<ReadLoopReadRequest::FailureSteps> failure_steps)
+void ReadableStreamDefaultReader::read_all_chunks(GC::Ref<ReadAllOnChunkSteps> chunk_steps, GC::Ref<ReadAllOnSuccessSteps> success_steps, GC::Ref<ReadAllOnFailureSteps> failure_steps)
 {
     // AD-HOC: Some spec steps direct us to "read all chunks" from a stream, but there isn't an AO defined to do that.
-    //         We implement those steps by using the "read all bytes" definition, with a custom callback to receive
-    //         each chunk that is read.
+    //         We implement those steps by continuously making default read requests, which is an identity transformation,
+    //         with a custom callback to receive each chunk that is read. This is done until the controller signals
+    //         that there are no more chunks to consume.
+    //         This function is based on "read_all_bytes" above.
     auto& realm = this->realm();
-    auto& vm = realm.vm();
+
+    auto promise_capability = WebIDL::create_promise(realm);
+
+    WebIDL::react_to_promise(
+        promise_capability,
+        GC::create_function(heap(), [this, chunk_steps, success_steps, failure_steps](JS::Value chunk) -> WebIDL::ExceptionOr<JS::Value> {
```

Comment:
nit on the name `chunk`: We are actually receiving an iterator Object here, and the "chunk" is its value property. So I'd swap the names here: call the parameter here `value`, and its value `chunk` below.